src/components/UplaodingDisplay/UploadingDisplay.js

```diff
@@ -0,0 +1,254 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { withStripes } from '@folio/stripes/core';
+
+import FileItem from './components/FileItem';
+
+
+class UploadingDisplay extends Component {
+  static propTypes = {
+    stripes: PropTypes.object,
+    endpointFileDef: PropTypes.string,
+    endpointFileUpload: PropTypes.string,
+  };
+
+  static defaultProps = {
+    endpointFileDef: '/data-import/upload/definition',
+    endpointFileUpload: '/data-import/upload/file',
+  };
+
+  static mapFilesFromProps(props, reducer, initReducerValue = {}) {
+    if (!props.initState) return {};
+
+    return props.initState.reduce(reducer, initReducerValue);
+  }
+
+  static mapFilesFromPropsReducer(result, currentFile) {
+    const keyNameValue = currentFile.name + currentFile.lastModified;
+
+    currentFile.keyName = keyNameValue;
+    currentFile.currentUploaded = 0;
+    result[keyNameValue] = currentFile;
+
+    return result;
+  }
+
+  static addHeaders = (xhr, headersObj) => {
```

Comment:
Rename to be just `headers`. We should avoid naming arguments with type in it as it can be changed later.